### PR TITLE
12_As an administrator, I need to see the of archived announcements

### PR DESCRIPTION
**Pull request by Jean-Beru**
Issue #14

### CODE REVIEW

**Comment by ogizanagi** [comment]
You'll need to rebase this PR on master.

**Comment by Jean-Beru** [comment]
This PR has already been rebased. I try to know why...

**Comment by Jean-Beru** [comment]
Done :smile:

File: src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php [hunk 1]
```diff
@@ -70,13 +71,27 @@ public function filterCountryName($code)
...
     public function filterContractType($contract)
     {
         $contractTypes = Job::getContactTypes();
 
-        return isset($contractTypes[$contract]) ? $contractTypes[$contract] : null;
+        return isset($contractTypes[$contract]) ? $contractTypes[$contract] : $contract;
+    }
+
+    /**
+     * Filter status type by its name.
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php]: That doesn't give any clue on what it really does.

File: src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php [hunk 2]
```diff
@@ -70,13 +71,27 @@ public function filterCountryName($code)
...
     public function filterContractType($contract)
     {
         $contractTypes = Job::getContactTypes();
 
-        return isset($contractTypes[$contract]) ? $contractTypes[$contract] : null;
+        return isset($contractTypes[$contract]) ? $contractTypes[$contract] : $contract;
+    }
+
+    /**
+     * Filter status name by its database value.
```
Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php]: I stil don't understand. What does it returns ? Why are you talking about database value ?

Comment (Jean-Beru) [inline src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php]: Database value means a constant like `full_time`. Anyway, I use "Replace ... with a readable value" formulation now.

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php]: Keep in mind database value doesn't mean anything in your model.

What your filter does is indeed returning a readable value from a constant. The `Filter` action in your docblock doesn't mean anything either. We know it's a filter, but over all it's a method, and we don't know what it does.

**Comment by ogizanagi** [comment]
:+1:

File: src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php (hunk 2 above)
Comment (Jean-Beru) [inline src/SensioLabs/JobBoardBundle/Twig/JobBoardExtension.php]: OK, thanks for the advice :wink:

**Pull request merged**